### DISCUSSION TITLE
Support for RADIUS auth clients

### DISCUSSION THREAD

**Discussion by terafirmanz** [zitadel/zitadel#7591] [Ideas]
I know its a legacy protocol but it would be amazing if client support for RADIUS could be added. There are still many apps that use older protocols for various reasons and it would be nice to be able to manage them from the same SSO tool.

Even then adding MFA via OTP append to password etc.

Any chance this could be considered?

**Comment by hifabienne** [comment]
Thanks for bringing the idea up.
Our priorization is based on the needs of our customers.
If we see a lot of reactions on this discussion and upvotes we will have a closer look.
So far we didn't hear a lot of feedback about this topic.

**Reply by Reiner030** [reply]
I found this issue while searching howto implement the suggested Radius plugin to the service.
=>
But then your company employees/manager shouldn't mentioned it in reddit comments and other places **commonly** as the better alternative to Keycloak which has a public available and usable https://vzakharchenko.github.io/keycloak-radius-plugin/ compared to your service ;-) (the other parts are maybe better but not in my scope/checked).
=> Authentik seems for this case with Radius Support also a good OpenSource / Service Alternative with an Radius/LDAP provider and e.g. Azure/EntraID as source - directly or with LDAP provider and FreeRadius it could do also EAP-TTLS.

BTW:
Radius itself is used by nearly all ISPs (old: modem access, still DSL, cable, ...), VPN providers (if it'snot the zero trust model) , all network hardware suppliers (LAN, WIFI) supporting 802.1x Auth and more.
You can read the functionalities on keycloak-radius-plugin page above.
Bests :-)

**Reply by fforootd** [reply]
Thank you for this pointer! That a nice plugin.

Zitadel might not (yet) be a good alternative to kc when radius is a needed, that is true 😁 I guess it is only matter of time until someone builds a radius endpoint.

 There is though a bunch of other cases, especially in CIAM and B2B, where we certainly are a great if not better alternative 😁

**Comment by terion-name** [comment]
RADIUS in apps is not common, but for WiFi and networking it would be great

**Reply by fforootd** [reply]
Hm yeah 802.1x/tacacs and these things still rely on that.

ATM one could build a radius wrapper on top of our session-api 😁

**Comment by makuc** [comment]
Agreed, support for RADIUS would be amazing for networking. It might become the reason I'll have to go for Authentik. Because we could use CodeB + Zitadel (via RADIUS) for local windows login....what we could do with Authentik (haven't tried yet).
